### ISSUE TITLE
Choose a more reliable network clock source for the CI

### ISSUE THREAD

**Issue by kennytm** [rust-lang/rust#78977]
In #63511, we added a clock drift check using the Date: of `http://detectportal.firefox.com/success.txt` as reference. However, the reply is extremely unreliable recently (?), so we can't use it to detect any drift at all.

```console
$ date -u; curl -fs --head http://detectportal.firefox.com/success.txt | grep Date
Thu Nov 12 10:49:23 UTC 2020
Date: Wed, 11 Nov 2020 12:48:07 GMT

$ date -u; curl -fs --head http://detectportal.firefox.com/success.txt | grep Date
Thu Nov 12 10:49:27 UTC 2020
Date: Thu, 12 Nov 2020 00:26:09 GMT

$ date -u; curl -fs --head http://detectportal.firefox.com/success.txt | grep Date
Thu Nov 12 10:49:30 UTC 2020
Date: Wed, 11 Nov 2020 12:48:07 GMT

$ date -u; curl -fs --head http://detectportal.firefox.com/success.txt | grep Date
Thu Nov 12 10:49:32 UTC 2020
Date: Thu, 12 Nov 2020 00:26:09 GMT
```

**Comment by Enselic** [comment]
Triage: Is this still a problem?

**Comment by kennytm** [comment]
already fixed from #90928.
